### PR TITLE
feat(kad): remove deprecated public modules

### PR DESCRIPTION
**Pull request by thomaseizinger**
## Description

This patch removes the 3 out of 4 deprecated public modules. I've left `store` for now because we made some mistakes in declaring that. The items within `store` still need to be publicly visible but I haven't yet figured out a good way of exporting / naming them. Thus, I've left that to a follow-up PR.

Related: #3647.

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Review by mxinden — APPROVED** [review]

File: protocols/kad/src/lib.rs [hunk 1]
```diff
@@ -72,7 +48,10 @@ pub mod record {
 
 mod addresses;
 mod behaviour;
+mod handler;
 mod jobs;
+mod kbucket;
```
Comment (gurinderu) [inline protocols/kad/src/lib.rs:53]: Why is it not public ?
take a look into get_closest_local_peers signature, it has a kbucket::Key as a parameter

Comment (gurinderu) [reply protocols/kad/src/lib.rs:53]: and the same story for handler + ConnectionClosed event

Comment (thomaseizinger) [reply protocols/kad/src/lib.rs:53]: `Key` is publicly exposed as `KBucketKey`. Does that solve your issue? The handler is an implementation detail. If you need to name it, you can use `<Kademlia as NetworkBehaviour>::ConnectionHandler`.

Comment (gurinderu) [reply protocols/kad/src/lib.rs:53]: yep, sound good, thanks

**Pull request merged**